### PR TITLE
fix: webhook message editing on forums missing thread_id

### PR DESCRIPTION
**Pull request by qoft**
## Summary

Fix editing a webhook message if the thread is a forum post or if the thread is a private thread (no starter message)

Issue [here](https://github.com/Pycord-Development/pycord/issues/1979)

## Information

- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [x] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
- [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [ ] I have updated the changelog to include these changes.

### CODE REVIEW

**Review by Lulalaby — APPROVED** [review]

File: discord/webhook/async_.py [hunk 1]
```diff
@@ -892,6 +892,8 @@ async def edit(
             thread = Object(self._thread_id)
         elif isinstance(self.channel, Thread):
             thread = Object(self.channel.id)
+        elif isinstance(self.channel, ForumChannel):
```
Comment (Lulalaby) [inline discord/webhook/async_.py:895]: The ci says ForumChannel can't be found

**Review by Lulalaby — CHANGES_REQUESTED** [review]

**Comment by qoft** [comment]
Changes done

**Comment by qoft** [comment]
bump

**Comment by Lulalaby** [comment]
bump deez nuts

**Comment by qoft** [comment]
changes were done already ☝😲

**Comment by Lulalaby** [comment]
We check through prs if we have time.
No need to bump anything.

**Review by plun1331 — CHANGES_REQUESTED** [review]
Please add a changelog entry.

**Comment by Lulalaby** [comment]
@qoft please address the requested changes

**Review by Lulalaby — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Review by ChickenDevs — APPROVED** [review]

**Pull request merged**